foundations/src/telemetry/server.rs

```diff
@@ -1,31 +1,140 @@
 #[cfg(feature = "metrics")]
 use super::metrics;
 use super::settings::TelemetrySettings;
+use crate::telemetry::log;
 use crate::BootstrapResult;
 use anyhow::Context as _;
 use futures_util::future::{BoxFuture, FutureExt};
-use hyper::server::conn::{AddrIncoming, AddrStream};
+use futures_util::{pin_mut, ready};
+use http_body_util::combinators::BoxBody;
+use http_body_util::{BodyExt, Empty, Full};
+use hyper::body::{Bytes, Incoming};
 use hyper::service::Service;
-use hyper::{header, Body, Method, Request, Response, Server, StatusCode};
+use hyper::{header, Method, Request, Response, StatusCode};
+use hyper_util::rt::TokioIo;
 use percent_encoding::percent_decode_str;
 use socket2::{Domain, SockAddr, Socket, Type};
 use std::collections::HashMap;
 use std::convert::Infallible;
-use std::future::{ready, Future, Ready};
-use std::net::{SocketAddr, TcpListener};
+use std::future::Future;
+use std::net::SocketAddr;
 use std::pin::Pin;
 use std::sync::Arc;
 use std::task::{Context, Poll};
+use tokio::net::TcpListener;
+use tokio::sync::watch;
 
-pub(super) type TelemetryServerFuture = Server<AddrIncoming, Router>;
+pub(super) struct TelemetryServerFuture {
```

Comment:
Let's move it in a separate module file

Reply:
I split `server.rs` into `server/mod.rs` and `server/router.rs`.